Cubical/Codata/Conat/Properties.agda

```diff
@@ -145,3 +147,11 @@ module Bisimulation where
   path≡bisim : ∀ {x y} → (x ≡ y) ≡ (x ≈ y)
   path≡bisim = ua path≃bisim
 
+  isProp≈ : ∀ {x y} → isProp (x ≈ y)
+  prove (isProp≈ p q i) = isProp≈′ (prove p) (prove q) i
```

Comment:
Can't you prove it instead by combining `isSetConat` and `path≡bisim` using `transport`?

Reply:
Answering my own question:

```agda
  isProp≈ : ∀ {x y} → isProp (x ≈ y)
  isProp≈ p q =
    p               ≡⟨ sym (iso p) ⟩
    misib (bisim p) ≡⟨ cong misib (isSetConat _ _ _ _) ⟩
    misib (bisim q) ≡⟨ iso q ⟩
    q               ∎
```

Reply:
Is there some synonyms that can do all these (`p < sym (bla p) > orz < something isSetSth > zro < bla q > q`)?

Reply:
More generally:

```agda

  SectionToProp-isProp :
    ∀ {a b} {A : Set a} {B : Set b} (f : A → B) g →
    section f g → isProp A → isProp B
  SectionToProp-isProp f g fg propA x y =
    x       ≡⟨ sym (fg x) ⟩
    f (g x) ≡⟨ cong′ f (propA _ _) ⟩
    f (g y) ≡⟨ fg y ⟩
    y ∎

  isProp≈ : ∀ {x y} → isProp (x ≈ y)
  isProp≈ {x} {y} = SectionToProp-isProp misib bisim iso (isSetConat x y)
```

Reply:
I like this! I'll put it into the Equiv module.

Reply:
We should have a function that proves "equivalence on hSet is hProp".. I'm trying to find an existing one in the library.

Reply:
Finally get a simplest proof: ` subst isProp path≡bisim (isProp[Set≡] isSetConat)`

Reply:
> We should have a function that proves "equivalence on hSet is hProp".. I'm trying to find an existing one in the library.

I don't understand what you're looking for... Regardless, the library is missing a lot of theory about HLevels/NTypes and equivalences. I was going to make an issue about this, but I forgot.

Reply:
> More generally:
> 
> ```agda
>   SectionToProp-isProp :
>     ∀ {a b} {A : Set a} {B : Set b} (f : A → B) g →
>     section f g → isProp A → isProp B
>   SectionToProp-isProp f g fg propA x y =
>     x       ≡⟨ sym (fg x) ⟩
>     f (g x) ≡⟨ cong′ f (propA _ _) ⟩
>     f (g y) ≡⟨ fg y ⟩
>     y       ∎
> 
>   isProp≈ : ∀ {x y} → isProp (x ≈ y)
>   isProp≈ = SectionToProp-isProp misib bisim iso (isSetConat _ _)
> ```

Nice, this should be included in `Foundations.HLevels`. I would restate it using retraction instead though. The general form of the lemma says that any retract `A` of an n-type `B` is an n-type (i.e. is of h-level n+2). This can of course be stated using a section instead, but at least in my head it is clearer what is going on if I think of it as a retraction as it says that `A` is a nice subtype of `B`.

Reply:
> I don't understand what you're looking for...

It's just a simple `\p -> p _ _` 😉